### PR TITLE
Fix refs to old K8s releases on glossary entries

### LINKED ISSUE #19535: Dated glossary entry for cloud controller manager
**This is a Bug Report**

**Problem:**
https://k8s.io/docs/reference/glossary/?all=true#term-cloud-controller-manager is out of date; it talks about Kubernetes releases v1.6 and v1.8

**Proposed Solution:**
At a minimum, remove the reference to releases v1.6 and v1.8 from the glossary entry.
It's OK to revise further as well.

**Page to Update:**
https://k8s.io/docs/reference/glossary/

**Additional Information:**
/kind cleanup
/help
/language en
/priority important-longterm

Also see issue #18086

### PR DESCRIPTION
**Pull request by davivcgarcia**
This PR replace references to old Kubernetes releases and API states.

Closes #19535 

/kind cleanup
/language en
/size S

### CODE REVIEW

File: content/en/docs/reference/glossary/cloud-controller-manager.md [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+ Cloud Controller Manager is the preferred way to integrate Kubernetes with any cloud infrastructure.
 
 <!--more--> 
 
-Kubernetes v1.6 contains a new binary called cloud-controller-manager. cloud-controller-manager is a daemon that embeds cloud-specific control loops. These cloud-specific control loops were originally in the kube-controller-manager. Since cloud providers develop and release at a different pace compared to the Kubernetes project, abstracting the provider-specific code to the cloud-controller-manager binary allows cloud vendors to evolve independently from the core Kubernetes code.
+Kubernetes contains a binary called cloud-controller-manager. cloud-controller-manager is a daemon that embeds cloud-specific control loops. These cloud-specific control loops were originally in the kube-controller-manager. Since cloud providers develop and release at a different pace compared to the Kubernetes project, abstracting the provider-specific code to the cloud-controller-manager binary allows cloud vendors to evolve independently from the core Kubernetes code.
```
[the remark this answers is not in the document]
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/cloud-controller-manager.md:19]: I wasn't sure if a content refactoring would require a new Issue, so I just changed the things mentioned in the original Issue. But you are absolutely right, the text needs to be revised and I'll do that!

[the remark this answers is not in the document]
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/cloud-controller-manager.md]: Please let me know if it got better now! :smile:

File: content/en/docs/reference/glossary/cloud-controller-manager.md [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+  Cloud Controller Manager is a Kubernetes component that embeds cloud-specific control logics.
 
 aka: 
 tags:
 - core-object
 - architecture
 - operation
 ---
- Cloud Controller Manager is an alpha feature in 1.8. In upcoming releases it will be the preferred way to integrate Kubernetes with any cloud.
+ Cloud Controller Manager is a Kubernetes component that embeds cloud-specific control logics.
```
Comment (kbhawkey) [inline content/en/docs/reference/glossary/cloud-controller-manager.md:15]: nit: s/logic/logics/g ?

File: content/en/docs/reference/glossary/host-aliases.md [hunk 3]
```diff
@@ -14,4 +14,4 @@ tags:
 
 <!--more-->
 
-[HostAliases](/docs/reference/generated/kubernetes-api/v1.13/#hostalias-v1-corev) is an optional list of hostnames and IP addresses that will be injected into the Pod's hosts file if specified. This is only valid for non-hostNetwork Pods.
+[HostAliases](https://kubernetes.io/docs/reference/generated/kubernetes-api/{{< latest-version >}}/#hostalias-v1-core) is an optional list of hostnames and IP addresses that will be injected into the Pod's hosts file if specified. This is only valid for non-hostNetwork Pods.
```
Comment (sftim) [inline content/en/docs/reference/glossary/host-aliases.md:17]: I'd use:
```suggestion
[HostAliases](https://kubernetes.io/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#hostalias-v1-core) is an optional list of hostnames and IP addresses that will be injected into the Pod's hosts file if specified. This is only valid for non-hostNetwork Pods.
```

so that if `{{< latest-version >}}` refers at some point to v1.19, the v1.18 docs still signpost readers to the v1.18 API docs.

Comment (davivcgarcia) [reply content/en/docs/reference/glossary/host-aliases.md:17]: Got it! I'm gonna change that.

File: content/en/docs/reference/glossary/host-aliases.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 title: HostAliases
 id: HostAliases
 date: 2019-01-31
-full_link: /docs/reference/generated/kubernetes-api/v1.13/#hostalias-v1-core
+full_link: /docs/reference/generated/kubernetes-api/{{< latest-version >}}/#hostalias-v1-core
```
[the remark this answers is not in the document]
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/host-aliases.md:5]: Got it! I'm gonna change that.

File: content/en/docs/reference/glossary/cloud-controller-manager.md [hunk 5]
```diff
@@ -1,19 +1,19 @@
 ---
 title: Cloud Controller Manager
 id: cloud-controller-manager
-date: 2018-04-12
+date: 2020-04-06
 full_link: /docs/tasks/administer-cluster/running-cloud-controller/
 short_description: >
-  Cloud Controller Manager is an alpha feature in 1.8. In upcoming releases it will be the preferred way to integrate Kubernetes with any cloud.
+  Cloud Controller Manager is a Kubernetes component that embeds cloud-specific control logics.
```
[the remark this answers is not in the document]
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/cloud-controller-manager.md:7]: Good to know! I'm gonna check that too.

File: content/en/docs/reference/glossary/cloud-controller-manager.md [hunk 6]
```diff
@@ -1,19 +1,19 @@
 ---
 title: Cloud Controller Manager
 id: cloud-controller-manager
-date: 2018-04-12
+date: 2020-04-06
```
[the remark this answers is not in the document]
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/cloud-controller-manager.md:4]: I thought this date was something like "last update". I'm gonna revert that change!

File: content/en/docs/reference/glossary/cloud-controller-manager.md (hunk 2 above)
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/cloud-controller-manager.md:15]: Got it! (I need to enable spellcheck on my editor ASAP... :stuck_out_tongue:)

File: content/en/docs/reference/glossary/host-aliases.md [hunk 7]
```diff
@@ -14,4 +14,4 @@ tags:
 
 <!--more-->
 
-[HostAliases](/docs/reference/generated/kubernetes-api/v1.13/#hostalias-v1-corev) is an optional list of hostnames and IP addresses that will be injected into the Pod's hosts file if specified. This is only valid for non-hostNetwork Pods.
+[HostAliases](https://kubernetes.io/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#hostalias-v1-core) is an optional list of hostnames and IP addresses that will be injected into the Pod's hosts file if specified. This is only valid for non-hostNetwork Pods.
```
Comment (kbhawkey) [inline content/en/docs/reference/glossary/host-aliases.md:17]: @davivcgarcia , What do you think about using,
/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#hostalias-v1-core

File: content/en/docs/reference/glossary/cloud-controller-manager.md [hunk 8]
```diff
@@ -4,16 +4,16 @@ id: cloud-controller-manager
...
 - architecture
 - operation
 ---
- Cloud Controller Manager is an alpha feature in 1.8. In upcoming releases it will be the preferred way to integrate Kubernetes with any cloud.
+ Cloud Controller Manager is a Kubernetes component that embeds cloud-specific control logic.
 
 <!--more--> 
 
-Kubernetes v1.6 contains a new binary called cloud-controller-manager. cloud-controller-manager is a daemon that embeds cloud-specific control loops. These cloud-specific control loops were originally in the kube-controller-manager. Since cloud providers develop and release at a different pace compared to the Kubernetes project, abstracting the provider-specific code to the cloud-controller-manager binary allows cloud vendors to evolve independently from the core Kubernetes code.
+Originally part of the kube-controller-manager, these cloud specific control logic abstract the provider code to the Cloud Controller Manager and enable cloud providers to develop and release at a different pace compared to the Kubernetes project.
```
Comment (kbhawkey) [inline content/en/docs/reference/glossary/cloud-controller-manager.md:19]: @sftim, This is quite different from your edits in pull request #19705.
Line 19 is hard to parse.

Comment (sftim) [reply content/en/docs/reference/glossary/cloud-controller-manager.md:19]: @davivcgarcia - I'd be happy to collaborate on getting the best bits of this PR and #19705 

(it's been a while since I sent #19705 in, and as far as I know that PR was good to merge, but it wasn't for me to review).

File: content/en/docs/reference/glossary/host-aliases.md (hunk 7 above)
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/host-aliases.md:17]: Oh my! Sorry my mistake, I don't know why I replaced it with the full URL. I'll fix that right away!

File: content/en/docs/reference/glossary/cloud-controller-manager.md (hunk 8 above)
Comment (davivcgarcia) [reply content/en/docs/reference/glossary/cloud-controller-manager.md:19]: @kbhawkey : what about `Originally part of the kube-controller-manager, the cloud-controller-manager is responsible to decoupling the interoperability between Kubernetes and the underlying cloud infrastructure, enabling cloud providers to release features at a different pace compared to the main project.` ?

**Comment by kbhawkey** [comment]
Thanks @davivcgarcia for your patience.
Lets get these changes in. There is a possibility that the cloud-controller-manager definition may change with updates from PR #19705.
/lgtm
